k8s_pod_filter.go

```diff
@@ -27,9 +27,9 @@ type K8sPodsMetadata struct {
 	} `json:"items"`
 }
 
```

Comment:
You want to define an interface here like:
```suggestion

type PodFilter interface {
     ShouldTailLogs(pod *Pod) (bool, error)
}

var _ PodFilter = &AnnotationPodFilter
var _ PodFilter = &StubPodFilter
```